test/e2e/e2e.go

```diff
@@ -188,14 +189,48 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 
 })
 
+var lock sync.Mutex
+var cleanupActions = map[*int]func(){}
+
+func AddCleanupAction(fn func()) *int {
+	p := new(int)
+	lock.Lock()
+	defer lock.Unlock()
+	cleanupActions[p] = fn
+	return p
+}
+
+func RemoveCleanupAction(p *int) {
+	lock.Lock()
+	defer lock.Unlock()
+	delete(cleanupActions, p)
+}
+
+func RunCleanupActions() {
+	list := []func(){}
+	func() {
+		lock.Lock()
+		defer lock.Unlock()
+		for _, fn := range cleanupActions {
+			list = append(list, fn)
+		}
+	}()
+	// Run unlocked.
+	for _, fn := range list {
+		fn()
+	}
+}
+
 // Similar to SynchornizedBeforeSuite, we want to run some operations only once (such as collecting cluster logs).
 // Here, the order of functions is reversed; first, the function which runs everywhere,
 // and then the function that only runs on the first Ginkgo node.
 var _ = ginkgo.SynchronizedAfterSuite(func() {
 	// Run on all Ginkgo nodes
-
 }, func() {
 	// Run only Ginkgo on node 1
+
+	RunCleanupActions()
```

Comment:
this will need to run on all nodes, since they are actually separate processes. (move to the other block.)